### PR TITLE
Add Family Code API

### PR DESCRIPTION
**Pull request by JoeTonDev**
I added Family Codes API grouped under 'accounts'

### CODE REVIEW

File: src/accounts.rs [hunk 1]
```diff
@@ -42,6 +50,17 @@ pub(crate) fn cancel_positions(client: &Client) -> Result<(), Error> {
     Ok(())
 }
 
+/// Determine whether an account exists under an account family and find the account family code.
+pub(crate) fn family_codes(client: &Client) -> Result<impl Iterator<Item = FamilyCode> + '_, Error> {
+    client.check_server_version(server_versions::ACCOUNT_SUMMARY, "It does not support fammily code requests.")?;
```
Comment (wboayue) [inline src/accounts.rs:55]: typo family

Comment (wboayue) [reply src/accounts.rs:55]: Server version for this should be `server_version::REQ_FAMILY_CODES`

File: src/messages.rs [hunk 2]
```diff
@@ -65,7 +65,7 @@ pub enum IncomingMessages {
     SecurityDefinitionOptionParameter = 75,
     SecurityDefinitionOptionParameterEnd = 76,
     SoftDollarTier = 77,
-    FamilyCodes = 78,
+    FamilyCode = 78,
```
Comment (wboayue) [inline src/messages.rs:68]: Should keep this `FamilyCodes` for consistency with IB source.

File: src/messages.rs [hunk 3]
```diff
@@ -95,6 +95,7 @@ pub enum IncomingMessages {
     WshEventData = 105,
     HistoricalSchedule = 106,
     UserInfo = 107,
+    FamilyCodeEnd = 108,
```
Comment (wboayue) [inline src/messages.rs:98]: I don't see this code defined in IB source

File: src/accounts/decoders.rs [hunk 4]
```diff
@@ -38,6 +38,17 @@ pub(crate) fn position(message: &mut ResponseMessage) -> Result<Position, Error>
     Ok(position)
 }
 
+pub(crate) fn family_code(message: &mut ResponseMessage) -> Result<FamilyCode, Error> {
```
Comment (wboayue) [inline src/accounts/decoders.rs:41]: This should return a vector similar to the contract_descriptions message https://github.com/wboayue/rust-ibapi/blob/723634d2d03bf91fc15c57c7610cd7cb6938529b/src/contracts/decoders.rs#L127.

This is a single message. Which contains a count for the number of family codes.

File: src/accounts/decoders.rs [hunk 5]
```diff
@@ -71,4 +82,19 @@ mod tests {
             assert!(false, "error decoding position: {err}");
         }
     }
+
+    #[test]
+    fn decode_family_codes() {
+        let mut message = super::ResponseMessage::from("0DU1236109\0F445566");
```
Comment (wboayue) [inline src/accounts/decoders.rs:88]: How did you get the sample response message? I usually start by creating an example for the API call. (e.g. https://github.com/wboayue/rust-ibapi/blob/main/examples/contract_details.rs). When we request is encoded correctly I can look at the logs to get a sample of the response message.

**Review by wboayue — CHANGES_REQUESTED** [review]
Thanks for the pull request. Overall looks good, left comments inline.
Main thing is fixing the decoding of the response message.

File: src/messages.rs (hunk 2 above)
Comment (JoeTonDev) [reply src/messages.rs:68]: Will change accordingly.  Thank you.

File: src/messages.rs (hunk 3 above)
Comment (JoeTonDev) [reply src/messages.rs:98]: Will remove code.

File: src/accounts/decoders.rs (hunk 4 above)
Comment (JoeTonDev) [reply src/accounts/decoders.rs:41]: Will do.  Thanks

File: src/accounts.rs (hunk 1 above)
Comment (JoeTonDev) [reply src/accounts.rs:55]: Will change accordingly.  Thanks

File: src/accounts/decoders.rs (hunk 5 above)
Comment (JoeTonDev) [reply src/accounts/decoders.rs:88]: I am stuck at this test.  How do I get the logs from the sample of the response message?
I've created a family_codes example: 
fn main() -> anyhow::Result<()> {
    let client = Client::connect("127.0.0.1:4002", 100).expect("connection failed");
 
    let family_codes = client.family_codes().expect("request failed");
    for family_code in family_codes {
        println!("{:4} {:4}", family_code[0].account_id, family_code[0].family_code)
    }

    Ok(())

}

Comment (JoeTonDev) [reply src/accounts/decoders.rs:88]: I am stuck at this problem.  Can you show me how you got the logs from 'positions' response message?

Comment (wboayue) [reply src/accounts/decoders.rs:88]: @JoeTonDev for this part you need an interactive broker paper trading account. With the IB desktop or gateway client running, I would create and example that uses your new code to fetch the family codes.

You can use [this example](https://github.com/wboayue/rust-ibapi/blob/main/examples/head_timestamp.rs) as a guide.

Notice the call to `env_logger::init`. This allows you to turn logging on/off using an environment variable. So you can go this to see the messages being sent and received from the IB server.

```bash
RUST_LOG=debug cargo run --example family_codes
```

**Comment by JoeTonDev** [comment]
New update pushed to 'add-family-codes'.

Thanks

**Review by wboayue — APPROVED** [review]
👍

**Pull request merged**